client/core/trade.go

```diff
@@ -1224,6 +1257,83 @@ func (t *trackedTrade) finalizeRedeemAction(match *matchTracker, coinID []byte)
 	return errs.ifAny()
 }
 
+func (t *trackedTrade) beginFindRedemption(match *matchTracker) {
+	swapCoinID := dex.Bytes(match.MetaData.Proof.TakerSwap)
+	fromAsset := t.wallets.fromAsset.ID
+	resultChan, err := t.wallets.fromWallet.FindRedemption(swapCoinID)
+	if err != nil {
+		log.Errorf("failed to start redemption search for match %v, %s contract %s: %v",
+			match.id, unbip(fromAsset), coinIDString(fromAsset, swapCoinID), err)
+		return
+	}
+	// Redemption search started, start goroutine to monitor result channel.
+	t.awaitingRedemption[swapCoinID.String()] = match.id
+	go t.waitForRedemption(resultChan)
+}
+
+// waitForRedemption waits on the passed `redemptionChan` for FindRedemption
+// results. Should be run as a goroutine.
+func (t *trackedTrade) waitForRedemption(redemptionChan chan *asset.FindRedemptionResult) {
+	frr := <-redemptionChan
+	if frr == nil {
+		return // channel closed without data
+	}
```

Comment:
Hmm. Is the idea to keep `Core` alive until the channels are all closed?

If so, we can have `trackedTrade.wg` that can be incremented in `trade.go` where necessary such as when a goroutine is started to monitor a redemption channel.
Then wait on all `trackedTrade.wg`s in `core.Run` after core's ctx is canceled.
```
	c.wg.Add(1)
	go func() {
		defer c.wg.Done()
		c.latencyQ.Run(ctx)
	}()
	c.wg.Wait()
	log.Infof("Waiting for trade trackers")
	c.connMtx.RLock()
	for _, dc := range c.conns {
		dc.tradeMtx.RLock()
		for _, trade := range dc.trades {
			trade.wg.Wait()
		}
		dc.tradeMtx.RUnlock()
	}
	c.connMtx.RUnlock()
```

If passing a context means waiting for the context's shutdown and then closing the find redemption channels, that might lead to issues when the asset tries to close the channel of it the asset already closed the channel. Safer to have the channels closed from one place - the sending end rather than the receiving end.